app/models/monthly_statistics/by_course_age_group.rb

```diff
@@ -0,0 +1,87 @@
+module MonthlyStatistics
+  class ByCourseAgeGroup
+    def table_data
+      {
+        rows: rows,
+        column_totals: column_totals_for(rows),
+      }
+    end
+
+  private
+
+    def rows
+      @rows ||= formatted_group_query.map do |age_group, statuses|
+        {
+          'Age group' => age_group,
+          'Recruited' => recruited_count(statuses),
+          'Conditions pending' => pending_count(statuses),
+          'Received an offer' => offer_count(statuses),
+          'Awaiting provider decisions' => awaiting_decision_count(statuses),
+          'Unsuccessful' => unsuccessful_count(statuses),
+          'Total' => statuses_count(statuses),
+        }
+      end
+    end
+
+    def column_totals_for(rows)
+      _age_group, *statuses = rows.first.keys
+
+      statuses.map do |column_name|
+        column_total = rows.inject(0) { |total, hash| total + hash[column_name] }
+        column_total
+      end
+    end
+
+    def formatted_group_query
+      counts = {
+        'Primary' => {},
+        'Secondary' => {},
+        'Further education' => {},
+      }
+
+      group_query.map do |item|
+        level, status = item[0]
+        count = item[1]
+        counts[level].merge!({ status => count })
+      end
+
+      counts
+    end
+
+    def group_query
+      ApplicationChoice
+        .joins(:course)
+        .where(current_recruitment_cycle_year: RecruitmentCycle.current_year)
+        .group('courses.level', 'status')
+        .count
+    end
+
+    def recruited_count(statuses)
```

Comment:
is it worth memoising these as they're called twice?

Reply:
I think this is ok as `statuses` is being passed in as a param plus memoization is set here so not to trigger more than one db query - https://github.com/DFE-Digital/apply-for-teacher-training/pull/5591/files#diff-95801d7d1eb4be97b00432a344a88fda8d6e7a03b64871ea344f8148cd23fcb3R13

Reply:
oh right sorry missed that!